engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -775,6 +775,54 @@ SELECT ?obsId {
 `, context)))).resolves.toHaveLength(1);
       });
     });
+
+    describe('commutative count', () => {
+      it('should count commutatively', async() => {
+        const context: QueryStringContext = {
+          sources: [
+            {
+              type: 'serialized',
+              value: `
+              @prefix ex: <http://example.org/> .
+            
+              <http://foo.org/id/graph/foo> {
+                <http://foo.org/id/object/foo>
+                  ex:foo ex:Foo ;
+                  ex:bar [] ;
+                  ex:baz "baz1", "baz2", "baz3" .
+              }
+              `,
+              mediaType: 'text/turtle',
+              baseIRI: 'http://example.org/',
+            },
+          ],
+        };
+
+        const received = (await arrayifyStream(await engine.queryBindings(`
+        SELECT (COUNT(DISTINCT ?s) AS ?subjects) (COUNT(DISTINCT ?o) as ?objects) ?p
+        FROM <http://foo.org/id/graph/foo>
+        {
+          ?s ?p ?o .
+        }
+        GROUP BY ?p
+        `, context))).sort((a, b) => a - b);
+
+        const expected = (await arrayifyStream(await engine.queryBindings(`
+        SELECT (COUNT(DISTINCT ?o) as ?objects) (COUNT(DISTINCT ?s) AS ?subjects) ?p
+        FROM <http://foo.org/id/graph/foo>
+        {
+          ?s ?p ?o .
+        }
+        GROUP BY ?p
+        `, context))).sort((a, b) => a - b);
+
+        expect(received).toHaveLength(expected.length);
+        for (const [ index, receivedBindings ] of received.entries()) {
```

Comment:
I'm not really sure I understand what this code does. Are you still working with bindings objects?
I'm not convinced this is actually correct 😅 

I think we can just manually do something like `expect(expected[0]).toEqual(received[0]);` and so on.

Reply:
expected and received are indeed bindings objects
I wanted to make sure the order in which these bindings objects are returned doesn't matter within the test, which is why I tried to sort the array of bindings objects in `received` and `expected`. For now the test succeeds without sorting as well.
(I sorted the entries in each bindings object according to the keys `subject` `object` and `predicate` because this order differs between these queries.)

Reply:
Let me attempt to clarify my issue with this test:
At the moment it is difficult to intuitively understand the expected output from this test.
What IMO would clarify things, is to just directly test all expected bindings (there should only be a few I guess) separately to precise values.
This would allow the code reader to see exactly what the expected output should be.

Reply:
Sorry for the misunderstanding. I've added an explicit test result, I wanted to compare both queries to the same variable, that way it would be clear at first glance that both queries in the test should yield the same result. So that explains why I sorted the query results.